src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnHeapHnsw.java

```diff
@@ -293,6 +295,30 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
                                                                                   postingsMap.keySet().size(), vectorValues.size());
         logger.debug("Writing graph with {} rows and {} distinct vectors", postingsMap.values().stream().mapToInt(VectorPostings::size).sum(), vectorValues.size());
 
+        final BiMap<Integer, Integer> newOrdinals = HashBiMap.create();
```

Comment:
More of a nit: Since ordinals will always have to be between `0, ..., n-1` before and after permuting (`n` = number of ordinals), we can just use `int[n]`s for the re-map and its inverse, where the array index serves as the natural key. 

This avoids any boxing of the `VectorPostings.getOrdinal()` value I see below, as well as hashing.

Reply:
It was a balance between creating array of numOrdinals size all the time vs BiMap populated only if there is a mapping change (getOrDefault takes care of rowId == ordinal cases).